### ISSUE TITLE
Difficulty applying to other areas

### ISSUE THREAD

**Issue by burgerberlin** [haoqiwang/vim#4]
Hello dear author, thank you very much for your work, very efficient. I applied your method on another pre-trained model, but I found that the effect was very poor. I know this is not your responsibility, but I still hope you can help me out.
My current situation is as follows, I still use a pre-trained Vit model, but texture, SVHN, LSUN and places365 all have AUROC close to 1 with cifar100 as ID, but only about 0.91 on cifar10 as OOD detection dataset . We ablated the Residual module in your method and found that this was mainly caused by the very large change in this score.
Since our current method does not involve bias b, I directly set b = np.ones(class_num). In addition, I also have a scal parameter that acts on logit. We use this parameter while calculating energy, and energy is normal, but in the Residual part we found It is better not to use this parameter. My question is: Is the current poor performance due to problems with b and scal parameters, or is it caused by different pre-trained models?
Looking forward to your reply, thank you very much

**Comment by innerlee** [comment]
Thanks for your interest in the method! Without details, I guess the bias and scaling is the culprit.

> Since our current method does not involve bias b, I directly set b = np.ones(class_num). In addition, I also have a scal parameter that acts on logit

Could please explain in detail what "our current method does not involve bias b"? Do you mean the bias on the last classification fc layer? In this case, if there is no b, then you can simply set it to 0.

Could you also explain in detail what "have a scal parameter that acts on logit" means? If you scale the logit by a constant, then you use the scaled logit in the computation of matching coefficient https://github.com/haoqiwang/vim/blob/master/benchmark.py#L247C5-L247C11 . The core idea is to match the scale of virtual logit and normal logit before the softmax.

Please let me know if there are more questions.

**Comment by burgerberlin** [comment]
I think I should have done these details right, but the effect is still not good. Thank you very much for your answer. I will think of a solution myself. If I have any questions, I will contact you again. Thank you.

**Comment by innerlee** [comment]
I'm happy to help.
